google/cloud/firestore_v1/stream_generator.py

```diff
@@ -14,21 +14,54 @@
 
 """Classes for iterating over stream results for the Google Cloud Firestore API.
 """
+from __future__ import annotations
 
 from collections import abc
+from typing import TYPE_CHECKING, Generator, Optional, TypeVar
+
+from google.cloud.firestore_v1.query_profile import (
+    ExplainMetrics,
+    QueryExplainError,
+)
+
+if TYPE_CHECKING:  # pragma: NO COVER
+    from google.cloud.firestore_v1.query_profile import ExplainOptions
+
+
+T = TypeVar("T")
 
 
 class StreamGenerator(abc.Generator):
-    """Generator for the streamed results."""
+    """Generator for the streamed results.
 
-    def __init__(self, response_generator):
```

Comment:
FYI: https://github.com/googleapis/python-firestore/issues/947

Maybe we should look at this as part of the changes?

Reply:
Thanks, I think with the mypy change this should be resolved, although there are still many type annotation errors remaining.